gstreamer-pbutils/src/encoding_profile.rs

```diff
@@ -0,0 +1,574 @@
+use auto::EncodingVideoProfile;
+use auto::EncodingContainerProfile;
+
+trait EncodingProfileEdit {
+    fn set_allow_dynamic_output(&self, allow_dynamic_output: bool);
+
+    fn set_description<'a, P: Into<Option<&'a str>>>(&self, description: P);
+
+    fn set_enabled(&self, enabled: bool);
+
+    fn set_format(&self, format: &gst::Caps);
+
+    fn set_name<'a, P: Into<Option<&'a str>>>(&self, name: P);
+
+    fn set_presence(&self, presence: u32);
+
+    fn set_preset<'a, P: Into<Option<&'a str>>>(&self, preset: P);
+
+    fn set_preset_name<'a, P: Into<Option<&'a str>>>(&self, preset_name: P);
+
+    fn set_restriction<'a, P: Into<Option<&'a gst::Caps>>>(&self, restriction: P);
+}
+
+impl<O: IsA<EncodingProfile> + IsA<glib::object::Object>> EncodingProfileEdit for O {
+
+    fn set_allow_dynamic_output(&self, allow_dynamic_output: bool) {
+        unsafe {
+            ffi::gst_encoding_profile_set_allow_dynamic_output(self.to_glib_none().0, allow_dynamic_output.to_glib());
+        }
+    }
+
+    fn set_description<'a, P: Into<Option<&'a str>>>(&self, description: P) {
+        let description = description.into();
+        let description = description.to_glib_none();
+        unsafe {
+            ffi::gst_encoding_profile_set_description(self.to_glib_none().0, description.0);
+        }
+    }
+
+    fn set_enabled(&self, enabled: bool) {
+        unsafe {
+            ffi::gst_encoding_profile_set_enabled(self.to_glib_none().0, enabled.to_glib());
+        }
+    }
+
+    fn set_format(&self, format: &gst::Caps) {
+        unsafe {
+            ffi::gst_encoding_profile_set_format(self.to_glib_none().0, format.to_glib_none().0);
+        }
+    }
+
+    fn set_name<'a, P: Into<Option<&'a str>>>(&self, name: P) {
+        let name = name.into();
+        let name = name.to_glib_none();
+        unsafe {
+            ffi::gst_encoding_profile_set_name(self.to_glib_none().0, name.0);
+        }
+    }
+
+    fn set_presence(&self, presence: u32) {
+        unsafe {
+            ffi::gst_encoding_profile_set_presence(self.to_glib_none().0, presence);
+        }
+    }
+
+    fn set_preset<'a, P: Into<Option<&'a str>>>(&self, preset: P) {
+        let preset = preset.into();
+        let preset = preset.to_glib_none();
+        unsafe {
+            ffi::gst_encoding_profile_set_preset(self.to_glib_none().0, preset.0);
+        }
+    }
+
+    fn set_preset_name<'a, P: Into<Option<&'a str>>>(&self, preset_name: P) {
+        let preset_name = preset_name.into();
+        let preset_name = preset_name.to_glib_none();
+        unsafe {
+            ffi::gst_encoding_profile_set_preset_name(self.to_glib_none().0, preset_name.0);
+        }
+    }
+
+    fn set_restriction<'a, P: Into<Option<&'a gst::Caps>>>(&self, restriction: P) {
+        let restriction = restriction.into();
+        let restriction = restriction.to_glib_none();
+        unsafe {
+            ffi::gst_encoding_profile_set_restriction(self.to_glib_none().0, restriction.0);
```

Comment:
As you say, this is `transfer full`. The solution here would be to use `restriction.to_glib_full()`, which is giving a new reference

Reply:
And the C code is wrong and should do
```c
g_return_if_fail (restriction == NULL || GST_IS_CAPS (restriction));
```
so that it actually allows `NULL` caps. Maybe as a workaround we can convert `None` here to `gst::Caps::new_any()` (and fix it on the C side).

Reply:
Basically this here
```diff
diff --git a/gstreamer-pbutils/src/encoding_profile.rs b/gstreamer-pbutils/src/encoding_profile.rs
index 1ad7d4b..d3768a1 100644
--- a/gstreamer-pbutils/src/encoding_profile.rs
+++ b/gstreamer-pbutils/src/encoding_profile.rs
@@ -2,6 +2,7 @@
 use gst;
 use glib;
 use ffi;
+use gst_ffi;
 
 use std::error;
 use std::fmt;
@@ -96,9 +97,13 @@ impl<O: IsA<EncodingProfile> + IsA<glib::object::Object>> EncodingProfileEdit fo
 
     fn set_restriction<'a, P: Into<Option<&'a gst::Caps>>>(&self, restriction: P) {
         let restriction = restriction.into();
-        let restriction = restriction.to_glib_none();
         unsafe {
-            ffi::gst_encoding_profile_set_restriction(self.to_glib_none().0, restriction.0);
+            let restriction = match restriction {
+                Some(restriction) => restriction.to_glib_full(),
+                None => gst_ffi::gst_caps_new_any(),
+            };
+
+            ffi::gst_encoding_profile_set_restriction(self.to_glib_none().0, restriction);
         }
     }
 }
```
Fixes the warning at least

Reply:
Changed the code for this. Thanks for the diff. Will push a patch for the C code as well.